### PR TITLE
Add org.v1993.NXDumpClient

### PR DESCRIPTION
**Pull request by v1993**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- - `--device=all` is required for USB access to Switch; user is prompted to manually install relevant udev rules on first launch
- - `--filesystem=xdg-download` is the default dump destination; alternate directories can be selected and use directory access portal
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/v1993/nxdumpclient
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- - Registered `v1993.org` (served by github pages), DNS changes will take a while to propagate.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by v1993** [comment]
bot, build org.v1993.NXDumpClient

File: modules/blueprint-compiler.json
Comment (bbhtt) [inline modules/blueprint-compiler.json:1]: Add to the main manifest

File: modules/libgusb.json
Comment (bbhtt) [inline modules/libgusb.json:1]: Add to the main manifest

File: modules/libportal.json
Comment (bbhtt) [inline modules/libportal.json:1]: Add to the main manifest

File: modules/libusb.json
Comment (bbhtt) [inline modules/libusb.json:1]: Add to the main manifest

This is in shared-modules

File: modules/libgusb.json
Comment (v1993) [reply modules/libgusb.json:1]: Is this required? I find it much nicer to manage modules this way.

File: org.v1993.NXDumpClient.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+app-id: org.v1993.NXDumpClient
```
Comment (bbhtt) [inline org.v1993.NXDumpClient.yml:1]: v1993.org doesn't seem to resolve, it has to be a domain under the control of the project/developer

File: modules/libgusb.json
Comment (bbhtt) [reply modules/libgusb.json:1]: Multiple moving parts make it much harder to manage.

File: org.v1993.NXDumpClient.yml (hunk 1 above)
Comment (v1993) [reply org.v1993.NXDumpClient.yml:1]: I've mentioned this in description. I suppose I can acquire in now if it's a hard requirement, although I'd prefer to be sure that it is first.

Comment (bbhtt) [reply org.v1993.NXDumpClient.yml:1]: Yes it is a blocker

Comment (bbhtt) [reply org.v1993.NXDumpClient.yml:1]: You can always use `io.github.foo.Foo` if you don't want to acquire the domain

File: modules/libgusb.json
Comment (hfiguiere) [reply modules/libgusb.json:1]: > Is this required? I find it much nicer to manage modules this way.

No it is not. It's way harder to do anything. Unless it's a shared-module or an auto-generated manifest, it should be inline.

File: org.v1993.NXDumpClient.yml (hunk 1 above)
Comment (hfiguiere) [reply org.v1993.NXDumpClient.yml:1]: **one doesn't simply invent a domain for their app-id.**

use a github associated app-id.

Comment (hfiguiere) [reply org.v1993.NXDumpClient.yml:1]: > I've mentioned this in description. I suppose I can acquire in now if it's a hard requirement, although I'd prefer to be sure that it is first.

you don't need to buy a domain. Just don't use one you don't own or for which there is no delegation. That's why you can use `io.github`.

File: modules/libgusb.json
Comment (v1993) [reply modules/libgusb.json:1]: I see; will change. I'm a bit skeptical on having one big file, but I have pretty little experience with flatpak.

**Comment by v1993** [comment]
I've merged modules into the main manifest and am in the process of setting up v1993.org (DNS changes take a while to propagate). Is there anything else to do in the meantime?

Also, should I setup `nxdumpclient.v1993.org` as well before acceptance of this PR? Or can that be done later when I do verification?

File: org.v1993.NXDumpClient.yml (hunk 1 above)
Comment (v1993) [reply org.v1993.NXDumpClient.yml:1]: > **one doesn't simply invent a domain for their app-id.**

Unless one does :upside_down_face:. I acquired ownership of v1993.org and put up a stub page there using GitHub pages. Is this sufficient for now?

**Comment by hfiguiere** [comment]
bot, build org.v1993.NXDumpClient

**Comment by v1993** [comment]
bot, build org.v1993.NXDumpClient

File: org.v1993.NXDumpClient.yml (hunk 1 above)
Comment (bbhtt) [reply org.v1993.NXDumpClient.yml:1]: Yes

**Pull request closed without merging**